rules/aip0122/resource_id_output_only.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0122
+
+import (
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/stoewer/go-strcase"
+	"google.golang.org/genproto/googleapis/api/annotations"
+)
+
+var resourceIdOutputOnly = &lint.FieldRule{
+	Name: lint.NewRuleName(122, "resource-id-output-only"),
+	OnlyIf: func(f *desc.FieldDescriptor) bool {
+		var idName string
+		p := f.GetParent().(*desc.MessageDescriptor)
+
+		// Build an expected ID field name based on the Resource `singular`
+		// field or by parsing the `type`.
+		isRes := utils.IsResource(p)
```

Comment:
nit: this could be reduced to just `utils.GetResource(p)` with a nil pointer check?

Reply:
I thought about that and then kept it this way for a few reasons:
1. because this API exists to use and consolidate that `!= nil` check in case we ever need to refactor that
2. because we use the `isRes` later on in the `return`, outside of the `if isRes`, so makes sense to keep a local variable

So I will keep it as is for now.